src/frontend_cache/cache.c

```diff
@@ -25,9 +54,46 @@ void* search_fast_cache(size_t size) {
     return NULL;
 }
 
+void* search_unsorted_cache(size_t size) {
+    t_chunk* unsorted_cache = g_pagemap->frontend_cache->unsorted_cache;
+    while (unsorted_cache) {
+        printf("unsorted_cache->size: %zu\n", unsorted_cache->size);
+        if (unsorted_cache->size >= size) {
+            return unsorted_cache;
+        }
+        unsorted_cache = unsorted_cache->fd;
+    }
+
+    return NULL;
+}
+
+
+void* search_sorted_cache(size_t size) {
+    char key[32];
+    snprintf(key, sizeof(key), "%zu", size);
+
+    cache_table* cache_table = g_pagemap->frontend_cache->cache_table;
```

Comment:
still need to determine how the pagetype fits with the cache_table, thoughts? @rswhipple

Reply:
I think it's mostly aspirational for this implementation. With a next version we could implement tailored algorithms to handle small vs large chunks. But, for this version, we could at minimum have them creating new chunks from the correct heap (small vs large)?